wagtail/admin/ui/tables/__init__.py

```diff
@@ -424,12 +426,23 @@ def __init__(
             self.classname = classname
         self.base_attrs = attrs or {}
 
+    def get_caption(self):
+        return self.caption
+
     def get_context_data(self, parent_context):
         context = super().get_context_data(parent_context)
         context["table"] = self
         context["request"] = parent_context.get("request")
         return context
 
+    @property
+    def has_caption(self):
+        return True if self.get_caption() else False
+
+    @property
+    def table_caption(self):
+        return self.get_caption()
+
```

Comment:
I think maybe should keep this simple. 
Remove the has_caption and property table_caption

The indirection is a bit confusing and doesn't really have a big improvement.